client.go

```diff
@@ -697,20 +708,88 @@ func (c *InternalHTTPClient) RestoreFrom(ctx context.Context, r io.Reader, index
 		}
 
 		// Parse slice from entry name.
-		slice, err := strconv.ParseUint(hdr.Name, 10, 64)
-		if err != nil {
-			return fmt.Errorf("invalid backup entry: %s", hdr.Name)
-		}
-
-		// Read file into buffer.
-		var buf bytes.Buffer
-		if _, err := io.CopyN(&buf, tr, hdr.Size); err != nil {
-			return err
-		}
-
-		// Restore file to all nodes that own it.
-		if err := c.restoreSliceFrom(ctx, buf.Bytes(), index, frame, view, slice); err != nil {
-			return err
+		log.Println("READING", hdr.Name)
+		parts := strings.Split(hdr.Name, "/")
+		if parts[0] == "schema" {
```

Comment:
I don't follow.  Are you saying don't include the schema in the backup file?

Reply:
no, i mean supply the entire schema to the server and have the server handle the creation of everything. so it would be one call to pilosa with a payload of schema (in protobuf format). we have schema-creating logic in other places, so this would be a research task to find out where/how to consolidate those.